service/controller/appcatalog/resource/appcatalogentry/resource.go

```diff
@@ -77,3 +89,65 @@ func (r *Resource) getCurrentEntryCRs(ctx context.Context, cr v1alpha1.AppCatalo
 
 	return currentEntryCRs, nil
 }
+
+func (r *Resource) getIndex(ctx context.Context, storageURL string) (index, error) {
+	indexURL := fmt.Sprintf("%s/index.yaml", strings.TrimRight(storageURL, "/"))
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("getting index.yaml from %#q", indexURL))
+
+	// We use https in catalog URLs so we can disable the linter in this case.
+	resp, err := http.Get(indexURL) // #nosec
+	if err != nil {
+		return index{}, microerror.Mask(err)
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return index{}, microerror.Mask(err)
+	}
+
+	var i index
+
+	err = yaml.Unmarshal(body, &i)
+	if err != nil {
+		return i, microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("got index.yaml from %#q", indexURL))
+
+	return i, nil
+}
+
+func equals(current, desired *v1alpha1.AppCatalogEntry) bool {
+	if current.Name != desired.Name {
+		return false
+	}
+	if !reflect.DeepEqual(current.Labels, desired.Labels) {
+		return false
+	}
+
+	if current.Spec.DateCreated.Unix() != desired.Spec.DateCreated.Unix() {
```

Comment:
line number 142 can't compare the same thing here? and what's the reason assign the nil after comparison?

Reply:
Yes this was weird. When I was running the operator with go run and kind I got timezone differences.

Using the unix epoch works but is more complex. I'll take another look and add some UTs for this function.

Reply:
Using reflect.DeepEqual for the date comparison doesn't work due to time zones. I thought it might just be locally but on ghost there was the same problem and all CRs were updated each loop.

Comparing using the unix epoch is reliable. We can then clear the date fields and use reflect.DeepEqual for the other fields.

I added a comment. PTAL. Also let me know if you have a better idea.